### ISSUE TITLE
`WinScrolled` not triggered when scrolling non-current window

### ISSUE THREAD

**Issue by noscript** [vim/vim#10235]
### Steps to reproduce

1. Run vim:
```sh
vim -g --clean +"autocmd WinScrolled * echo expand('<amatch>') localtime()" +"help" +"split"
```
2. Scroll the current window with mouse wheel. Window ID will be printed at the bottom.
3. Move the mouse cursor to another window, but do not click it. Scroll with the mouse. Window ID is not printed.

### Expected behaviour

Window ID of the window under mouse is printed.

### Version of Vim

8.2.4794

### Environment

OS: Debian 11
Terminal: GUI

**Comment by LemonBoy** [comment]
The fix is easy, writing a test not so much (?):

```patch
diff --git a/src/mouse.c b/src/mouse.c
index 42963341c..ceee52756 100644
--- a/src/mouse.c
+++ b/src/mouse.c
@@ -1127,6 +1127,8 @@ ins_mousescroll(int dir)
 	}
 #endif
 	did_scroll = TRUE;
+
+	may_trigger_winscrolled();
     }
 
     curwin->w_redr_status = TRUE;
@@ -2088,6 +2090,8 @@ nv_mousescroll(cmdarg_T *cap)
 	redraw_for_cursorline(curwin);
 # endif
 
+    may_trigger_winscrolled();
+
     curwin->w_redr_status = TRUE;
 
     curwin = old_curwin;
```

**Comment by brammool** [comment]
> The fix is easy, writing a test not so much (?):
> 
> ```patch
> diff --git a/src/mouse.c b/src/mouse.c
> index 42963341c..ceee52756 100644
> --- a/src/mouse.c
> +++ b/src/mouse.c
> @@ -1127,6 +1127,8 @@ ins_mousescroll(int dir)
>  	}
>  #endif
>  	did_scroll = TRUE;
> +
> +	may_trigger_winscrolled();
>      }
>  
>      curwin->w_redr_status = TRUE;
> @@ -2088,6 +2090,8 @@ nv_mousescroll(cmdarg_T *cap)
>  	redraw_for_cursorline(curwin);
>  # endif
>  
> +    may_trigger_winscrolled();
> +
>      curwin->w_redr_status = TRUE;
>  
>      curwin = old_curwin;
> ```

I wonder if this causes trouble for those who use the WinScrolled event
without checking the window ID.  I guess not, since the event itself
does not say how much was scrolled, thus when checking the current
window it would be clear that there was no scroll.

Would be nice to have a test.

-- 
hundred-and-one symptoms of being an internet addict:
29. Your phone bill comes to your doorstep in a box.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
fixed by 66e13aedc7986c83644d537a8fdd3cb006507678
